File: plugins/Tester/incompleteUV.py
import medic
from maya import OpenMaya


class IncompleteUV(medic.PyTester):
    def __init__(self):
        super(IncompleteUV, self).__init__()

    def Name(self):
        return "IncompleteUV"

    def Description(self):
        return "Mesh(s) has incomplete UV(s)"

    def Match(self, node):
        return node.object().hasFn(OpenMaya.MFn.kMesh)

    def test(self, node):
        mesh = None
        try:
            mesh = OpenMaya.MFnMesh(node.object())
        except:
            return None

        uv_count = 0

        map_list = []
        mesh.getUVSetNames(map_list)

        for i, mp in enumerate(map_list):
            n_uvs = mesh.numUVs(mp)

            if i == 0:
                uv_count = n_uvs
            elif uv_count != n_uvs:
                return medic.PyReport(node)

        if uv_count is 0:
            return medic.PyReport(node)

        return None


def Create():
    return IncompleteUV()

File: plugins/Tester/incompleteUV.cpp
#include "medic/pluginapi.h"
#include <maya/MFn.h>
#include <maya/MStatus.h>
#include <maya/MObject.h>
#include <maya/MItMeshFaceVertex.h>
#include <maya/MFnMesh.h>
#include <maya/MStringArray.h>
#include <maya/MFnSingleIndexedComponent.h>


using namespace MEDIC;


class IncompleteUV : public MdTester
{
    std::string Name();
    std::string Description();
    bool Match(MdNode *node);
    MdReport *test(MdNode *node);
};


std::string IncompleteUV::Name()
{
    return "IncompleteUV";
}

std::string IncompleteUV::Description()
{
    return "Vertex has no UVs";
}

bool IncompleteUV::Match(MdNode *node)
{
    return node->object().hasFn(MFn::kMesh);
}

MdReport *IncompleteUV::test(MdNode *node)
{
    MStatus stat;
    bool error = false;
    MStringArray uvsets;
    MFnSingleIndexedComponent comp;
    MObject comp_obj = comp.create(MFn::kMeshVertComponent);

    MItMeshFaceVertex it(node->object(), &stat);

    if (stat == MS::kFailure)
    {
        return 0;
    }

    MFnMesh mesh(node->object(), &stat);
    if (stat == MS::kFailure)
    {
        return 0;
    }

    mesh.getUVSetNames(uvsets);

    if (uvsets.length() < 1)
    {
        return new MdReport(node);
    }

    while (!it.isDone())
    {
        for (unsigned int i = 0; i < uvsets.length(); ++i)
        {
            if (!it.hasUVs(uvsets[i]))
            {
                error = true;
                comp.addElement(it.vertId());
                break;
            }
        }

        it.next();
    }

    if (!error)
    {
        return 0;
    }

    return new MdReport(node, comp_obj);
}


MEDIC_PLUGIN_API MdTester *Create()
{
    return new IncompleteUV();
}
